internal/declarative/v2/reconciler.go

```diff
@@ -215,7 +229,9 @@ func (r *Reconciler) initialize(obj Object) error {
 		status.Synced = []shared.Resource{}
 	}
 
-	if status.State == "" {
+	if !obj.GetDeletionTimestamp().IsZero() {
+		status.State = shared.StateDeleting
```

Comment:
why we need to set deleting state manually here in the  `initialize` step? What happens if manifest CR is under deleting, but have warning or error state before? In the initialize step, we should not change original state, only add if there is no state.